src/components/Router/index.js

```diff
@@ -155,8 +155,13 @@ class MainRouter extends Component {
     return {steps, token, language, documentType, step: this.state.crossDeviceInitialStep, woopraCookie}
   }
 
-  onFlowChange = (newFlow, newStep, previousFlow, previousStep) => {
-    if (newFlow === "crossDeviceSteps") this.setState({crossDeviceInitialStep: previousStep})
+  onFlowChange = (newFlow, newStep, previousFlow, previousStep, previousStepType) => {
+    if (newFlow === "crossDeviceSteps") {
+      this.setState({
+        crossDeviceInitialStep: previousStep,
+        crossDeviceInitialStepType: previousStepType,
```

Comment:
what is the `previousStepType`?

Reply:
if we can easily access it and we have a function for it, might be better not to store in the state, since it's one less thing on our state.

Reply:
Definitely not a fan of storing more in state.

When flow changes, `componentsList` is built based on the new flow, so the previous `componentList` gets lost.

But on this new screen we need to know where it's coming from to display different copies, so we either have to:
- recreate the old `componentList` and pass it down somehow to `crossDevice/Intro` and access `oldComponentList[previousStep].type`
- store the current component type in state, in the same way we store `previousStep` and `previousFlow`

Reply:
hmm, so what is the `previousStep` isn't it the index? If so, at what time can it be used to get its corresponding value?

Reply:
`previousStep` holds the numeric index of the last step before changing flows. 

Even if we persist that index after changing the flow, we can't extract the step type `("document"|"face")`, because the `componentList` will then change:

Before changing the flow:
```
{component: {...}, {step: {type: "welcome", ...}},
{component: {...}, {step: {type: "document", ...}},
{component: {...}, {step: {type: "document", ...}},
{component: {...}, {step: {type: "document", ...}},
{component: {...}, {step: {type: "face", ...}}],
{component: {...}, {step: {type: "face", ...}}],
{component: {...}, {step: {type: "complete", ...}},
```

After changing the flow:
```
componentList = [
{component: {...}, step: {type: "crossDevice", ...}},
{component: {...}, {step: {type: "crossDevice", ...}},
{component: {...}, {step: {type: "crossDevice", ...}},
{component: {...}, {step: {type: "complete", ...}},
```

Just before changing the flow is the last time we can access `componentList` to inspect the current step's type.